state-chain/chains/src/eth.rs

```diff
@@ -77,6 +77,43 @@ impl EthereumTrackedData {
 	}
 }
 
+mod fees {
+	// TODO: refine these constants.
+	pub const BASE_COST_PER_BATCH: u128 = 10_000;
+	pub const GAS_COST_PER_FETCH: u128 = 40_000;
+	pub const GAS_COST_PER_TRANSFER: u128 = 50_000;
+}
+
+impl FeeEstimationApi<Ethereum> for EthereumTrackedData {
+	fn estimate_ingress_fee(
+		&self,
+		asset: <Ethereum as Chain>::ChainAsset,
+	) -> <Ethereum as Chain>::ChainAmount {
+		use fees::*;
+
+		// Note: this is taking the egress cost of the swap in the ingress currency (and basing the
+		// cost on the ingress chain).
+		let gas_cost_per_fetch = BASE_COST_PER_BATCH +
+			match asset {
+				assets::eth::Asset::Eth => Zero::zero(),
+				assets::eth::Asset::Flip | assets::eth::Asset::Usdc => GAS_COST_PER_FETCH,
+			};
```

Comment:
As discussed, this is a type, should be: GAS_COST_PER_BATCH + GAS_COST_PER_FETCH